### DISCUSSION TITLE
Collapse and expand python blocks

### DISCUSSION THREAD

**Discussion by mogplus88** [thonny/thonny#3347] [General]
Just for fun today I had a look at the Arduino python editor. It's quite good but not as comprehensive as Thonny, and I didn't like the way output was printed in the repl. 
Anyway, one feature I did like was the ability to collapse and expand python blocks. Easy and intuitive to use, and very helpful when trying to sort out a problem by hiding code that is not relevant to the problem, if you see what I mean. Very useful for checking alignment too, like all your ifs and elses, etc. 
Hiding code is useful too. When I used ISPF on IBM mainframes and dumb terminals, many decades ago, the editor had a feature that allowed chunks of code to be hidden, so that all the relevant bits of code you were working on could be displayed on the screen at the same time. (The block of code was replaced by a single line of hyphens.) Saves a lot of scrolling up and down. Some sort of split screen feature would work too I guess (ISPF had that feature as well BTW).
Just my two cents worth.
;-)

**Comment by aivarannamaa** [comment]
This is in the TODO list: #1090
